pkg/reconciler/common/releases.go

```diff
@@ -74,42 +74,53 @@ func InstalledManifest(instance v1alpha1.KComponent) (mf.Manifest, error) {
 	return fetch(installedManifestPath(current, instance))
 }
 
-// IsUpDowngradeEligible returns the bool indicate whether the installed manifest is able to upgrade or downgrade to
-// the target manifest.
-func IsUpDowngradeEligible(instance v1alpha1.KComponent) bool {
+// IsVersionValidMigrationEligible returns the bool indicate whether the target version is valid and the installed
+// manifest is able to upgrade or downgrade to the target manifest.
+func IsVersionValidMigrationEligible(instance v1alpha1.KComponent) error {
+	var errMess error
+	target := sanitizeSemver(TargetVersion(instance))
+	if !semver.IsValid(target) {
+		return fmt.Errorf("target version %v is not in a valid semantic versioning format.", target)
+	}
+
+	if len(strings.Split(target, ".")) < 2 {
+		return fmt.Errorf("target version %v should at least include the major and minor numbers.", target)
+	}
+
+	targetMinor, err := strconv.Atoi(strings.Split(target, ".")[1])
+	if err != nil {
+		return fmt.Errorf("minor number of the target version %v should be an integer.", target)
+	}
```

Comment:
If not sure if I understand that part. Why only checking the minor, and not the major and patch? Wouldn't the `semver` lib be able to validate the whole Semver string here?

Reply:
We check the major number first, then the minor.
these lines of code should be located downstairs.
i moved then upwards because i need to validate the target version.

Reply:
You're checking the minor here, not the major. The logic is

1. check minor
1. do something unrelated
1. check major

But why? I still don't get it.